ctypes.go

```diff
@@ -124,3 +132,48 @@ func cstring(name string) *byte {
 	copy(b, name)
 	return &b[0]
 }
+
+// cast is used to centralized unsafe use C allocated pointer.
+// We take the address and then dereference it to trick go vet from creating a possible misuse of unsafe.Pointer
+func cast[T any](ptr uintptr) *T {
```

Comment:
If you want to give it a try, I think this is the solution to accept pointers and knowing the pointer type:

```go
func cast[T any, PT interface { *T }](ptr uintptr) PT
```

Reply:
<img width="597" alt="image" src="https://github.com/DataDog/go-libddwaf/assets/47679741/fb30b96d-d6d1-438f-a46e-dcf3cedbd743">